packages/quantic/cypress/integration/case-classification/case-classification.cypress.ts

```diff
@@ -0,0 +1,385 @@
+      visitCaseClassification({
+        maxChoices: defaultMaxChoices,
+      });
+
+      scope('when loading the page', () => {
+        Expect.displayLabel(true);
+        Expect.numberOfInlineOptions(0);
+        Expect.numberOfSuggestions(0);
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+      });
+
+      scope('when fetching suggestions', () => {
+        const suggestionsCount = 2;
+        mockCaseClassification(
+          defaultField,
+          allOptions.slice(0, suggestionsCount)
+        );
+        interceptClassificationsIndefinitely();
+        fetchClassifications();
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+        Expect.numberOfSuggestions(0);
+        Expect.numberOfInlineOptions(0);
+        Expect.displayLoading(true);
+      });
+    });
+  });
+
+  describe('when no suggestions are found', () => {
+    it('should display only the select dropdown', () => {
+      visitCaseClassification({
+        maxChoices: defaultMaxChoices,
+      });
+
+      scope('when loading the page', () => {
+        Expect.numberOfInlineOptions(0);
+        Expect.numberOfSuggestions(0);
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+      });
+
+      scope('when fetching suggestions', () => {
+        mockCaseClassification(defaultField, []);
+        fetchClassifications();
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+        Expect.numberOfSuggestions(0);
+        Expect.numberOfInlineOptions(0);
+      });
+    });
+  });
+
+  describe('with maxChoices set to 1', () => {
+    it('should display only the select dropdown', () => {
+      visitCaseClassification({
+        maxChoices: 1,
+      });
+
+      scope('when loading the page', () => {
+        Expect.displayLabel(true);
+        Expect.numberOfInlineOptions(0);
+        Expect.numberOfSuggestions(0);
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+      });
+
+      scope('when fetching suggestions', () => {
+        const suggestionsCount = 2;
+        mockCaseClassification(
+          defaultField,
+          allOptions.slice(0, suggestionsCount)
+        );
+        fetchClassifications();
+        Expect.displaySelectTitle(false);
+        Expect.numberOfSuggestions(0);
+        Expect.numberOfInlineOptions(0);
+        Expect.displaySelectInput(true);
+      });
+
+      scope('when selecting an option from the select input', () => {
+        const clickedIndex = 3;
+
+        Actions.openSelectInput();
+        Actions.clickSelectOption(clickedIndex);
+        Expect.logUpdatedClassificationFromSelectOption(
+          defaultField,
+          clickedIndex
+        );
+      });
+    });
+  });
+
+  describe('with invalid number of maxChoices', () => {
+    it('should display only the select dropdown when maxChoices is set to 0', () => {
```

Comment:
Yes
> Exactly, in other words the minimum value that `maxChoices` can take is 1. The logic behind this is that max choices represents the maximum number of choices we propose to the user, and a single choice can be a suggestion, an inline option or the select drop down. so for example if maxChoices is set to 3 we can show 2 suggestions and the select drop down. if maxChoices is set to 1 we only can show the select drop down. Having maxChoices set to 0 means that we don't want anything no suggestion and no inline option and no select drop down and this will make our component kind of useless. so maxChoices is always greater or equal to 1. and in the case where the user gives maxChoices a negative value or 0, we always reset the value to 1.